keel/api/v1/chats/views.py

```diff
@@ -0,0 +1,95 @@
+from django.db.models import Q
+
+from rest_framework.viewsets import GenericViewSet
+from rest_framework import mixins, viewsets, status as HTTP_STATUS
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.exceptions import ValidationError
+from rest_framework.pagination import LimitOffsetPagination
+
+from keel.authentication.backends import JWTAuthentication
+from keel.Core.err_log import log_error
+from keel.Core.constants import GENERIC_ERROR
+from keel.chats.models import Chat, ChatRoom
+
+from .serializers import ChatListSerializer, ChatCreateSerializer
+
+
+class ChatList(GenericViewSet):
+
+    authentication_classes = [JWTAuthentication]
+    permission_classes = (IsAuthenticated,)
+
+    def listChats(self, request, format = 'json'):
+
+        response = {
+            "status" : 0,
+            "message" : "Chat list is fetched successfully",
+            "data" : [],
+        }
+
+        user = request.user
+        user_id = user.id
+
+        req_data = request.data.dict()
+        case_id = req_data.get("case_id","")
```

Comment:
use serializer to validate data
filter query doesn't raises a DoesNotExist exception
and try to perform al db operations in the viewset only!!

Reply:
you mean, Shall I create a separate `ChatRoomSerializer` to validate the case_id coming from the request body?

and DB query is :
`ChatRoom.objects.filter(Q(user = user_id) | Q(agent = user_id)).get(case = case_id)`

In the end, I have used `get`, thus using DoesNotExist

Reply:
if filter return None...will 'get' raise DoesNotExist ?

Reply:
Filter will always return Queryset, in case of no data, that wud be Empty QuerySet.
'get' on EmptyQuerySet will raise DoesNotExist